extension/wrapper.cpp

```diff
@@ -144,6 +168,8 @@ Satrec__sgp4(PyObject *self, PyObject *args)
 static PyMethodDef Satrec_methods[] = {
     {"twoline2rv", (PyCFunction)Satrec_twoline2rv, METH_VARARGS | METH_CLASS,
      PyDoc_STR("Initialize the record from two lines of TLE text.")},
+    {"sgp4init", (PyCFunction)Satrec_sgp4init, METH_VARARGS | METH_CLASS,
```

Comment:
I'll ask for one small pivot: let’s switch this from a constructor to a plain method. The underlying C++ API makes no restriction on how many times `sgp4init()` is called on a given struct, which allows re-use of an already allocated struct to represent a new (or even a slightly tweaked) satellite. So let’s expose that full power here. Happily, that will not only increase flexibility but should simplify the code a bit.

Reply:
Sorry - This is my first dive into CPython - I've reviewed the docs, but I'm not sure how to implement a "plain method" as opposed to a constructor -- however, agree with you that it would be convenient to call sgp4init on a modified set of variables, repeatedly, and would be glad for the more optimal implementation.  

Could you give this pivot request a go, please?

Reply:
First, try taking a look at `Satrec__sgp4()` — it’s a plain method taking arguments, exactly the plan you’ll want to use for the init method. My guess is that you’ll just return `None` from the init method since its result is written to the object itself; lots of web resources should be available on how to return that value, as it's a very popular one.

Reply:
Taking a look...

Reply:
Thanks! Given that constructors are actually more complicated than plain methods, my guess is that you'll be able to adapt to the plain-method pattern just fine, given how well you did the constructor.